### PR TITLE
Normalize message button in profile sidebar

### LINKED ISSUE #6410: Normalize Message button handling with remote authentication
From @rabuzarus in https://github.com/friendica/friendica/issues/3838#issuecomment-340836080

> Friendica should provide two different ways of handling the `Message` button.
> 
> If a person is a visitor (no established connection) the `Message` button on the profile page should point the visitor to `/wallmessage` (in frio it is opened in a modal). All messages which are written with `/wallmassege` are submitted as not validated author (because the system can't verify (at the moment) if the author of this wallmessage is really the author he/she claims to be. With other words, at the moment friendica is not able to authenticate contacts who aren't connected to the contact).
> 
> If a person is a guest (remote user - a contact with an established connection) the `Message` button on the profile page should redirect to the guests local `message` page (with additional data to tell the local `message` page the name and address of the recipient. The recipient should be the profile owner of the visited profile page). So it would be possible to send real private messages with authenticated contact information.
> 
> BTW: Frio can handle the two different situations (https://github.com/friendica/friendica/blob/develop/view/theme/frio/js/modal.js#L349)
> 
> 
> Some bug tracing:
> Which of the two links will be used is decided at https://github.com/friendica/friendica/blob/develop/include/identity.php#L268 (and L284).
> 
> When I visit the support forum as authenticated user (guest through magic-link) the `Message` button is pointing me to `/wallmessage/helpers`. But it should point to my local `message` module (because I'm a authenticated guest). Something went wrong here.
> 
> Because of the fact that I'm authenticated, no redirection url is submitted to the helpers forum (we don't need it because all data of an authenticated contact are stored in the contact table and we take all necessary date from this table) and `/wallmesage` will stop if we don't have an URL (https://github.com/friendica/friendica/blob/develop/mod/wallmessage.php#L77). In other themes `/wallmessage` will output an error message. Frio doesn't output this error message because the content of `/wallmessage` is loaded to a modal.
> 
> Back to the real problem? The big question is: Why do we don't get the redirection link to local `message` page for authenticated users? (https://github.com/friendica/friendica/blob/develop/include/identity.php#L268 and some lines below).
> 
> 
> Semi-OT:
> Is it useful to have the possibility to write private messages to forum accounts or should we remove it in such cases.

### PR DESCRIPTION
**Pull request by MrPetovan**
Fixes #6410

Phew!

It started as a simple task: showing the Message button under people's avatar in the sidebar for additional networks.

Then I found out about #6410 and long story short I completely rewrote the profile sidebar link conditions, and I added those links under the people's avatar in the contact context as well.

On to #7240

### CODE REVIEW

File: src/Model/Profile.php [hunk 1]
```diff
@@ -304,127 +303,102 @@ private static function sidebar($profile, $block = 0, $show_connect = true)
...
-				}
-			}
-		}
-
-		if (!$show_connect) {
-			$connect = false;
+		if (isset($profile['url'])) {
+			$profile_url = $profile['url'];
+		} else {
+			$profile_url = System::baseUrl() . '/profile/' . $profile['nickname'];
```
Comment (nupplaphil) [inline src/Model/Profile.php]:
```suggestion
			$profile_url = $a->getBaseURL() . '/profile/' . $profile['nickname'];
```

File: src/Model/Profile.php [hunk 2]
```diff
@@ -304,127 +303,102 @@ private static function sidebar($profile, $block = 0, $show_connect = true)
...
-		if (!$is_contact && $profile['uid'] == local_user() && Feature::isEnabled(local_user(), 'multi_profiles')) {
-			$profile['edit'] = [System::baseUrl() . '/profiles', L10n::t('Profiles'), '', L10n::t('Manage/edit profiles')];
-			$r = q(
-				"SELECT * FROM `profile` WHERE `uid` = %d",
-				local_user()
-			);
+		if (!$is_contact && $local_user_is_self) {
+			if (Feature::isEnabled(local_user(), 'multi_profiles')) {
+				$profile['edit'] = [System::baseUrl() . '/profiles', L10n::t('Profiles'), '', L10n::t('Manage/edit profiles')];
+				$r = q(
```
Comment (nupplaphil) [inline src/Model/Profile.php]:
```suggestion
				DBA::selectFirst('profile', [], ['uid' => local_user()]);
```

File: src/Model/Profile.php [hunk 3]
```diff
@@ -304,127 +303,102 @@ private static function sidebar($profile, $block = 0, $show_connect = true)
...
+							'alt' => L10n::t('Profile Image'),
+							'profile_name' => $rr['profile-name'],
+							'isdefault' => $rr['is-default'],
+							'visibile_to_everybody' => L10n::t('visible to everybody'),
+							'edit_visibility' => L10n::t('Edit visibility'),
+						];
+					}
 				}
+			} else {
+				$profile['edit'] = [System::baseUrl() . '/profiles/' . $profile['id'], L10n::t('Edit profile'), '', L10n::t('Edit profile')];
```
Comment (nupplaphil) [inline src/Model/Profile.php]:
```suggestion
				$profile['edit'] = [$a->getBaseURL() . '/profiles/' . $profile['id'], L10n::t('Edit profile'), '', L10n::t('Edit profile')];
```

**Comment by annando** [comment]
Since we are in the RC phase, I would suggest to apply as few changes as possible.

**Comment by MrPetovan** [comment]
I'm with @annando, the changes you are suggesting aren't even part of the initial issue. I'll keep them in mind for the future though,

**Comment by nupplaphil** [comment]
The changes are part of changes you made, f.e. you explicity introduced `System::baseUrl()`, so I thought you could have used `$a->getBaseURL()` instead too. But nevermind :)

**Comment by MrPetovan** [comment]
You're right for one line that I touched, the others are just indentation change and I'd rather not edit them further.

**Pull request merged**